src/hybrid/hybrid_request.c

```diff
@@ -56,9 +79,10 @@ static PLN_LoadStep *createImplicitLoadStep(void) {
  *
  * @param req The HybridRequest containing multiple AREQ search requests
  * @param params Pipeline parameters including aggregation settings and scoring context
+ * @param status QueryError pointer to store detailed error information on failure
  * @return REDISMODULE_OK on success, REDISMODULE_ERR on failure
  */
-int HybridRequest_BuildPipeline(HybridRequest *req, const HybridPipelineParams *params) {
+int HybridRequest_BuildPipeline(HybridRequest *req, const HybridPipelineParams *params, QueryError *status) {
```

Comment:
Why do we need the extra error, instead of working with the `HybridRequest` error (tail + branches)?